code/modules/jobs/access.dm

```diff
@@ -60,49 +60,37 @@
 		for(var/b in text2access(req_one_access_txt))
 			req_one_access += b
 
+// Check if an item has access to this object
 /obj/proc/check_access(obj/item/I)
+	return check_access_list(I ? I.GetAccess() : null)
+
+
+/obj/proc/check_access_list(list/access_list)
 	gen_access()
 
-	if(!istype(src.req_access, /list)) //something's very wrong
+	if(!istype(req_access, /list)) //something's very wrong
 		return TRUE
 
-	var/list/L = src.req_access
-	if(!L.len && (!src.req_one_access || !src.req_one_access.len)) //no requirements
+	if(!length(req_access) && !length(req_one_access))

```

Comment:
req_access.len, since you know it's a list by this point. LAZYLEN for req_one_access

Reply:
`LAZYLEN()` is just `length()`